indico/cli/i18n.py

```diff
@@ -113,62 +244,241 @@ def _wrapper(**kwargs):
+    def decorator(fn):
+        fn = click.option('--python', is_flag=True, help='i18n used in python and Jinja code.')(fn)
+        fn = click.option('--react', is_flag=True, help='i18n used in react code.')(fn)
+        if require_js:
+            fn = click.option('--javascript', is_flag=True, help='i18n used in javascript code.')(fn)
+        else:  # the only command that does not require js is compile_catalog which checks the string format at the end
+            fn = click.option('--no-check', is_flag=True, required=False,
+                              help='Skip running the string format validation.')(fn)
+        if require_locale:
+            fn = click.option('--locale', required=True)(fn)
+        if plugins:
+            fn = click.argument('plugin_dirs', required=True, type=click.Path(exists=True, file_okay=False,
+                                                                              resolve_path=True),
+                                nargs=-1, callback=_validate_plugin_dirs)(fn)
+        elif all_plugins:
+            fn = click.argument('plugin_dirs', type=click.Path(exists=True, file_okay=False, resolve_path=True),
+                                callback=_validate_all_plugins_dir)(fn)
+        return fn
+    return decorator
+
+
+def compile_catalog_react(directory=INDICO_DIR):
+    """Compile react catalogs."""
+    translations_dir = _get_translations_dir(directory)
+    try:
+        for locale in os.listdir(translations_dir):
+            po_file = os.path.join(translations_dir, locale, 'LC_MESSAGES', 'messages-react.po')
+            json_file = os.path.join(translations_dir, locale, 'LC_MESSAGES', 'messages-react.json')
+            if not os.path.exists(po_file):
+                continue
+            output = subprocess.check_output(['npx', 'react-jsx-i18n', 'compile', po_file])
+            json.loads(output)  # just to be sure the JSON is valid
+            with open(json_file, 'wb') as f:
+                f.write(output)
+    except subprocess.CalledProcessError as err:
+        click.secho('Error: ' + err, fg='red', bold=True, err=True)
+
+
+def extract_messages_react(directory=INDICO_DIR):
+    """Extract messagges for react."""
+    if directory is INDICO_DIR:
+        client_path = os.path.join('indico', 'web', 'client')
+        module_path = os.path.join('indico', 'modules')
+    else:
+        packages = [x for x in find_packages(directory) if '.' not in x]
+        assert len(packages) == 1
+        client_path = os.path.join(directory, packages[0], 'client')
+        module_path = os.path.join(directory, packages[0], 'modules')
+        if not os.path.exists(client_path) and not os.path.exists(module_path):
+            return
+    with _chdir(INDICO_DIR):
+        output = subprocess.check_output(['npx', 'react-jsx-i18n', 'extract', client_path, module_path],
+                                         env=dict(os.environ, FORCE_COLOR='1'))
```

Comment:
Sometimes just on of the folders exists an then you get this in the output for several plugins:
```
Invalid path: indico-plugins/cloud_captchas/indico_cloud_captchas/modules
```

Reply:
Yes this happens when they don't have the modules directory. I'm thinking of maybe only running this command for the clients folder for plugins and removing the modules folder since non of the plugins that _I'm aware_ of have been using react in the modules or even have it for that matter. Is this a good idea?

Reply:
Okay, it's now checking if the path exists.